src/main/java/seedu/kitchenhelper/notification/IngredientNotification.java

```diff
@@ -1,4 +1,118 @@
 package seedu.kitchenhelper.notification;
 
+import seedu.kitchenhelper.command.ListChoreCommand;
+import seedu.kitchenhelper.exception.KitchenHelperException;
+import seedu.kitchenhelper.object.Chore;
+import seedu.kitchenhelper.object.ingredient.Ingredient;
+import seedu.kitchenhelper.ui.Ui;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+
 public class IngredientNotification {
+
+    public static final String EXPIRING_NOTIFICATION = "These ingredients expiring date is approaching!\n";
+    public static final String LOWQUANTITY_NOTIFICATION = "These ingredients is currently low stock [< 5]!\n";
+    public static final String EXPIRED_NOTIFICATION = "These ingredients are expired!\n";
+    public static final String EMPTYEXPIRING_NOTIFICATION = "You have no ingredients that is "
+            + "expiring for the next 3 days.\n";
+    public static final String EMPTYLOWQUANTITY_NOTIFICATION = "You have no ingredients that is low in stock [< 5].\n";
+    public static final String EMPTYEXPIRED_NOTIFICATION = "You have no ingredients that is expired.\n";
+
+
+    public String getNotifications(ArrayList<Ingredient> ingredientArrayList) {
+        String notification = "";
+        ArrayList<Ingredient> ingredientExpiring = isExpiring(ingredientArrayList);
+        ArrayList<Ingredient> ingredientLowQuantity = isLowQuantity(ingredientArrayList);
+        ArrayList<Ingredient> ingredientExpired = isExpired(ingredientArrayList);
+
+
+        if (!ingredientExpiring.isEmpty()) {
+            notification += EXPIRING_NOTIFICATION;
+            for (Ingredient i : ingredientExpiring) {
+                notification += i.getIngredientName() + "|" + i.getQuantity() + "|"
+                        + i.getPrice() + "|" + i.getExpiryDate() + "\n";;
+            }
+        } else {
+            notification += EMPTYEXPIRING_NOTIFICATION;
+        }
+        if (!ingredientLowQuantity.isEmpty()) {
+            notification += LOWQUANTITY_NOTIFICATION;
+            for (Ingredient i : ingredientLowQuantity) {
+                notification += i.getIngredientName() + "|" + i.getQuantity() + "|"
+                        + i.getPrice() + "|" + i.getExpiryDate() + "\n";;
+            }
+        } else {
+            notification += EMPTYLOWQUANTITY_NOTIFICATION;
+        }
+        if (!ingredientExpired.isEmpty()) {
+            notification += EXPIRED_NOTIFICATION;
+            for (Ingredient i : ingredientExpired) {
+                notification += i.getIngredientName() + "|" + i.getQuantity() + "|"
+                        + i.getPrice() + "|" + i.getExpiryDate() + "\n";;
```

Comment:
Very clear and concise but maybe you would want to put this block into a method since its repeated 3x?

Reply:
okay. will do another commit. Do not merge till then